### PR TITLE
fix: Support python 3.6

### PR DESCRIPTION
**Pull request by friendtocephalopods**
### Summary of Changes
Modify workflow to support 3.6.

It works just fine with 3.6!
Fix some details from setup.py while I'm at it, and add types for the conftest file.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.

### CODE REVIEW

File: amundsen_gremlin/utils/streams.py [hunk 1]
```diff
@@ -354,6 +354,7 @@ def _consumer(things: Iterable[V], ignored: None) -> None:
     return _actual_state
 
 
+# NB: This will not work on python 3.6; requires 3.7 or later
```
Comment (alran) [inline amundsen_gremlin/utils/streams.py:357]: is this the only functionality that won't work on 3.6?

**Review by alran — APPROVED** [review]
Thanks for doing this! I wonder if it makes sense to write a short comment about the 3.6/3.7 divide in the README (but I think it shouldn't really impact people unless they use that one bit of code)

File: amundsen_gremlin/utils/streams.py (hunk 1 above)
Comment (friendtocephalopods) [reply amundsen_gremlin/utils/streams.py:357]: Yep! asyncio changed significantly between 3.6 and 3.7. 

Fortunately we don't use this in the proxy, and folks can use the synchronous one for their own databuilder if they cannot upgrade to 3.7

**Comment by friendtocephalopods** [comment]
> Thanks for doing this! I wonder if it makes sense to write a short comment about the 3.6/3.7 divide in the README (but I think it shouldn't really impact people unless they use that one bit of code)

Yeah, I need to write a comment about `--roundtrip` for pytest also, I'll do that in a subsequent PR.

**Pull request merged**